lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+RANDOM = Random.new;
+
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 1
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    if @alive
```

Comment:
Looking at the code that was already there (the missing apple case), we used the accessor through `self.any_apples?`. This seems to work just as well without the `self.`.

What's the preferred idiom?